### PR TITLE
Fix --purge / --purge-before-date dependency issue

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
Currently, the dependency check only works if `--purge` is specified first. If the `--purge` is after `--purge-before-date`, the dependency check action still report missing require args `--purge`. For example, `cou plan --purge-before-date 2000-1-1 --purge` will fail.

This PR fixed that issue, making the order of `--purge` and `--purge-before-date` unimportant

### CODE REVIEW

**Review by Pjack — DISMISSED** [review]

**Review by gabrielcocenza — DISMISSED** [review]

File: cou/commands.py [hunk 1]
```diff
@@ -185,8 +168,11 @@ def get_subcommand_common_opts_parser() -> argparse.ArgumentParser:
     )
     subcommand_common_opts_parser.add_argument(
         "--archive-batch-size",
-        help="Batch size for nova old database data archiving.\n"
-        "Decrease the batch size if performance issues are detected.\n(default: 1000)",
+        help=(
+            "Batch size for nova old database data archiving."
+            "\nDecrease the batch size if performance issues are detected.\n(default: 1000)"
+            "\nThis option requires --archive."
```
Comment (samuelallan72) [inline cou/commands.py:174]: Could you also add the logic to enforce this, copying the logic for purge/purge-before-date?

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
Could you improve the commit message and PR description wording?  It really confused me, because the wording and present tense suggested that the limitation applied to these changes.  But in fact this change fixes that limitation.

File: cou/commands.py (hunk 1 above)
Comment (samuelallan72) [reply cou/commands.py:174]: (Alternatively, that could be done in another PR, but perhaps the changes to --archive help message should be out of scope here then.)

Comment (chanchiwai-ray) [reply cou/commands.py:174]: Ah, sorry, this was accidentally added. This option does not required `--archive`

File: cou/commands.py [hunk 2]
```diff
@@ -555,6 +540,13 @@ def parse_args(args: Any) -> CLIargs:  # pylint: disable=inconsistent-return-sta
                     subparsers.choices["upgrade"].print_help()
             parser.exit()
 
+        # validate arguments
+        validation_errors = []
+        if parsed_args.purge_before and not getattr(parsed_args, "purge", None):
```
Comment (samuelallan72) [inline cou/commands.py:545]:
```suggestion
        if parsed_args.purge_before and not parsed_args.purge:
```

File: cou/commands.py [hunk 3]
```diff
@@ -555,6 +540,13 @@ def parse_args(args: Any) -> CLIargs:  # pylint: disable=inconsistent-return-sta
...
+            validation_errors.append("--purge-before-date requires --purge")
+        if len(validation_errors) > 0:
```
Comment (samuelallan72) [inline cou/commands.py:547]:
```suggestion
        if validation_errors:
```

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
Thanks @chanchiwai-ray , this looks pretty good.  I left some code lint comments though. :)

**Review by samuelallan72 — APPROVED** [review]

**Review by gabrielcocenza — APPROVED** [review]

File: cou/commands.py (hunk 3 above)
Comment (chanchiwai-ray) [reply cou/commands.py]: Fixed

File: cou/commands.py (hunk 2 above)
Comment (chanchiwai-ray) [reply cou/commands.py]: Fixed

**Pull request merged**